pkg/providers/ack/ack.go

```diff
@@ -105,9 +107,10 @@ func (p *DefaultProvider) GetClusterCNI(_ context.Context) (string, error) {
 
 func (p *DefaultProvider) GetNodeRegisterScript(ctx context.Context,
 	labels map[string]string,
+	nodeClaim *karpv1.NodeClaim,
```

Comment:
para labels can be extract from nodeClaim, let's only use para nodeClaim.

Reply:
we also need to pass `capacityType` as parameter even though the labels can be removed
```go
	labels := lo.Assign(nodeClaim.Labels, map[string]string{karpv1.CapacityTypeLabelKey: capacityType})
	userData, err := p.ackProvider.GetNodeRegisterScript(ctx, labels, nodeClaim, kubeletCfg)
```
Is that what we want?

Reply:
We can move:
```go
	labels := lo.Assign(nodeClaim.Labels, map[string]string{karpv1.CapacityTypeLabelKey: capacityType})
``` 

into this function, to extract the labels

Reply:
done